### ISSUE TITLE
Can't select my Organisation a second time.

### ISSUE THREAD

**Issue by th-uni** [eduvpn/python-eduvpn-client#384]
# Installation

sudo apt install apt-transport-https curl
$ curl -L https://app.eduvpn.org/linux/deb/eduvpn.key | sudo apt-key add -
$ echo "deb https://app.eduvpn.org/linux/deb/ stable main" | sudo tee -a /etc/apt/sources.list.d/eduvpn.list
$ sudo apt update
$ sudo apt install eduvpn-client

# Version

dpkg -l | grep eduvpn
ii  eduvpn-client                              2.0.0-1                                          all          VPN client for educational networks
ii  eduvpn-client-data                         2.0.0-1                                          all          VPN client for educational networks (data files)
ii  python3-eduvpn-client                      2.0.0-1                                          all          eduVPN client (Python 3)
dpkg -l | grep nacl
ii  python3-nacl                               1.1.2-1build1                                    amd64        Python bindings to libsodium (Python 3)

# OS/Distribution
$ lsb_release -a
No LSB modules are available.
Distributor ID: LinuxMint
Description:    Linux Mint 19.3 Tricia
Release:        19.3
Codename:       tricia
$ uname -a
Linux 5.0.0-32-generic #34~18.04.2-Ubuntu SMP Thu Oct 10 10:36:02 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

# Problem desciption:
It is a similar problem to this issues:
https://github.com/eduvpn/python-eduvpn-client/issues/372
https://github.com/eduvpn/python-eduvpn-client/issues/380

**eduvpn-gui**
With original Mint 19 Setup I was not able to discover my Organization, 
like issue/372.
But I was able to enter our eduvpn server manually and I was able to authorize and to connect to the vpn.
After closing the Client and reopen it, I now was not able to select it again.
It crashes with this output:
```
2021-06-23 09:12:26,947 - MainThread - INFO - eduvpn.ui.ui - ui.py:133 - i18n successfully initialized

(eduvpn-gui:5365): Gtk-WARNING **: 09:12:27.057: Overriding tab label for notebook

(eduvpn-gui:5365): Gtk-WARNING **: 09:12:27.059: Overriding tab label for notebook
2021-06-23 09:12:27,066 - MainThread - INFO - eduvpn.ui.ui - ui.py:172 - starting ui
2021-06-23 09:12:27,172 - MainThread - INFO - eduvpn.app - app.py:111 - network transitioning: <InitialNetworkState> -> found_previous_connection
2021-06-23 09:12:27,172 - MainThread - INFO - eduvpn.app - app.py:130 - interface transitioning: <InitialInterfaceState> -> no_active_connection_found
2021-06-23 09:12:27,179 - MainThread - INFO - eduvpn.app - app.py:141 - interface transitioned: no_active_connection_found -> <MainState servers=[<CustomServer 'https://edu-vpn.uni-erfurt.de/'>]>
2021-06-23 09:12:27,179 - MainThread - INFO - eduvpn.app - app.py:122 - network transitioned: found_previous_connection -> <DisconnectedState>
2021-06-23 09:12:27,181 - init-server-db - INFO - eduvpn.remote - remote.py:23 - Requesting https://disco.eduvpn.org/v2/server_list.json
2021-06-23 09:12:27,443 - init-server-db - INFO - eduvpn.remote - remote.py:37 - Retrieving signature https://disco.eduvpn.org/v2/server_list.json.minisig
2021-06-23 09:12:27,637 - init-server-db - INFO - eduvpn.remote - remote.py:44 - verifying signature of https://disco.eduvpn.org/v2/server_list.json
2021-06-23 09:12:27,639 - init-server-db - ERROR - eduvpn.remote - remote.py:76 - Got exception 'VerifyKey' object has no attribute '__unicode__' requesting https://disco.eduvpn.org/v2/server_list.json for server_list
Exception in thread init-server-db:
Traceback (most recent call last):
  File "/usr/lib/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.6/threading.py", line 864, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3/dist-packages/eduvpn/app.py", line 77, in initialize_server_db
    self.server_db.update()
  File "/usr/lib/python3/dist-packages/eduvpn/server.py", line 244, in update
    server_list = remote.list_servers(SERVER_URI)
  File "/usr/lib/python3/dist-packages/eduvpn/remote.py", line 74, in list_servers
    result = request(uri, verify=True)['server_list']
  File "/usr/lib/python3/dist-packages/eduvpn/remote.py", line 46, in request
    _ = validate(content=response.content, signature=signature)
  File "/usr/lib/python3/dist-packages/eduvpn/crypto.py", line 94, in validate
    logger.debug(f"Used signature {f}")
  File "/usr/lib/python3/dist-packages/nacl/utils.py", line 54, in __str__
    return self.__unicode__()
AttributeError: 'VerifyKey' object has no attribute '__unicode__'

2021-06-23 09:12:29,969 - MainThread - INFO - eduvpn.ui.ui - ui.py:548 - selection empty
2021-06-23 09:12:29,971 - MainThread - INFO - eduvpn.app - app.py:130 - interface transitioning: <MainState servers=[<CustomServer 'https://edu-vpn.uni-erfurt.de/'>]> -> connect_to_server
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/eduvpn/ui/ui.py", line 553, in on_select_server
    self.app.interface_transition('connect_to_server', server)
  File "/usr/lib/python3/dist-packages/eduvpn/app.py", line 134, in interface_transition
    transition, self, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/eduvpn/state_machine.py", line 139, in transition
    new_state = transition_func(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/eduvpn/interface/state.py", line 83, in connect_to_server
    return transition.connect_to_server(app, server)
  File "/usr/lib/python3/dist-packages/eduvpn/interface/transition.py", line 60, in connect_to_server
    metadata = storage.get_current_metadata(oauth_login_url)
  File "/usr/lib/python3/dist-packages/eduvpn/storage.py", line 77, in get_current_metadata
    created = datetime.fromisoformat(created)
AttributeError: type object 'datetime.datetime' has no attribute 'fromisoformat'
```
**After Upgrade python3-nacl**
```
apt install ./python3-nacl_1.3.0-5_amd64.deb
dpkg -l | grep nacl
ii  python3-nacl                               1.3.0-5                                          amd64        Python bindings to libsodium (Python 3)
```
**Clean eduvpn config**
`rm ~/.config/eduvpn/ -rf`

Now I was able to find my organisation and login to vpn. But after closing the application I was not able to connect again.
```2021-06-23 09:21:51,879 - MainThread - INFO - eduvpn.ui.ui - ui.py:133 - i18n successfully initialized

(eduvpn-gui:5812): Gtk-WARNING **: 09:21:52.001: Overriding tab label for notebook

(eduvpn-gui:5812): Gtk-WARNING **: 09:21:52.003: Overriding tab label for notebook
2021-06-23 09:21:52,009 - MainThread - INFO - eduvpn.ui.ui - ui.py:172 - starting ui
2021-06-23 09:21:52,134 - MainThread - INFO - eduvpn.app - app.py:111 - network transitioning: <InitialNetworkState> -> found_previous_connection
2021-06-23 09:21:52,134 - MainThread - INFO - eduvpn.app - app.py:130 - interface transitioning: <InitialInterfaceState> -> no_active_connection_found
2021-06-23 09:21:52,141 - MainThread - INFO - eduvpn.app - app.py:141 - interface transitioned: no_active_connection_found -> <MainState servers=[<InstituteAccessServer 'Universität Erfurt'>]>
2021-06-23 09:21:52,141 - MainThread - INFO - eduvpn.app - app.py:122 - network transitioned: found_previous_connection -> <DisconnectedState>
2021-06-23 09:21:52,143 - init-server-db - INFO - eduvpn.remote - remote.py:23 - Requesting https://disco.eduvpn.org/v2/server_list.json
2021-06-23 09:21:52,396 - init-server-db - INFO - eduvpn.remote - remote.py:37 - Retrieving signature https://disco.eduvpn.org/v2/server_list.json.minisig
2021-06-23 09:21:52,584 - init-server-db - INFO - eduvpn.remote - remote.py:44 - verifying signature of https://disco.eduvpn.org/v2/server_list.json
2021-06-23 09:21:52,588 - init-server-db - INFO - eduvpn.remote - remote.py:23 - Requesting https://disco.eduvpn.org/v2/organization_list.json
2021-06-23 09:21:52,948 - init-server-db - INFO - eduvpn.remote - remote.py:37 - Retrieving signature https://disco.eduvpn.org/v2/organization_list.json.minisig
2021-06-23 09:21:53,138 - init-server-db - INFO - eduvpn.remote - remote.py:44 - verifying signature of https://disco.eduvpn.org/v2/organization_list.json
2021-06-23 09:21:53,148 - MainThread - INFO - eduvpn.app - app.py:130 - interface transitioning: <MainState servers=[<InstituteAccessServer 'Universität Erfurt'>]> -> server_db_finished_loading
2021-06-23 09:21:53,149 - MainThread - INFO - eduvpn.app - app.py:141 - interface transitioned: server_db_finished_loading -> <MainState servers=[<InstituteAccessServer 'Universität Erfurt'>]>

###### first click on the organisation, nothing happens:
2021-06-23 09:22:17,702 - MainThread - INFO - eduvpn.ui.ui - ui.py:548 - selection empty
2021-06-23 09:22:17,703 - MainThread - INFO - eduvpn.app - app.py:130 - interface transitioning: <MainState servers=

###### second click on the organisation:
[<InstituteAccessServer 'Universität Erfurt'>]> -> connect_to_server
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/eduvpn/ui/ui.py", line 553, in on_select_server
    self.app.interface_transition('connect_to_server', server)
  File "/usr/lib/python3/dist-packages/eduvpn/app.py", line 134, in interface_transition
    transition, self, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/eduvpn/state_machine.py", line 139, in transition
    new_state = transition_func(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/eduvpn/interface/state.py", line 83, in connect_to_server
    return transition.connect_to_server(app, server)
  File "/usr/lib/python3/dist-packages/eduvpn/interface/transition.py", line 60, in connect_to_server
    metadata = storage.get_current_metadata(oauth_login_url)
  File "/usr/lib/python3/dist-packages/eduvpn/storage.py", line 77, in get_current_metadata
    created = datetime.fromisoformat(created)
AttributeError: type object 'datetime.datetime' has no attribute 'fromisoformat'
2021-06-23 09:23:26,823 - MainThread - INFO - eduvpn.ui.ui - ui.py:548 - selection empty
2021-06-23 09:23:26,823 - MainThread - INFO - eduvpn.app - app.py:130 - interface transitioning: <MainState servers=[<InstituteAccessServer 'Universität Erfurt'>]> -> connect_to_server
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/eduvpn/ui/ui.py", line 553, in on_select_server
    self.app.interface_transition('connect_to_server', server)
  File "/usr/lib/python3/dist-packages/eduvpn/app.py", line 134, in interface_transition
    transition, self, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/eduvpn/state_machine.py", line 139, in transition
    new_state = transition_func(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/eduvpn/interface/state.py", line 83, in connect_to_server
    return transition.connect_to_server(app, server)
  File "/usr/lib/python3/dist-packages/eduvpn/interface/transition.py", line 60, in connect_to_server
    metadata = storage.get_current_metadata(oauth_login_url)
  File "/usr/lib/python3/dist-packages/eduvpn/storage.py", line 77, in get_current_metadata
    created = datetime.fromisoformat(created)
AttributeError: type object 'datetime.datetime' has no attribute 'fromisoformat'
```

Now I am only able to connect to our vpn if I delete eduvpn profile befor every login
`rm ~/.config/eduvpn/ -rf`

**Comment by alvra** [comment]
Thanks for the detailed report. 
Looks like you've hit two bugs; #362 / #372, and one that doesn't have an issue yet but is being solved in #383.

**Comment by th-uni** [comment]
Thanks for the fix in #383

I've copied the new file **storage.py** (from github) over the old in
/usr/lib/python3/dist-packages/eduvpn/storage.py

Now I can select my organization and start the vpn. 

Many thanks.

**Comment by alvra** [comment]
Thanks for the confirmation, glad it helped! The fix will be in the next release.
